File: src/dashboard/src/types/feature.ts
export interface Feature {
  id: number
  name: string
  data_type: string
  source: string
  description?: string
  category?: string
  available: boolean
  min_value?: number
  max_value?: number
  mean_value?: number
  created_at: string
  updated_at?: string
}

export interface FeatureFilters {
  source?: string
  category?: string
  data_type?: string
  available?: boolean
}

File: src/backend/trading_server/src/domain/models/feature.py
"""
Feature metadata model for data sources
"""

from dataclasses import dataclass
from typing import Optional


@dataclass
class Feature:
    """Feature metadata for data sources"""

    name: str  # Unique feature identifier
    data_type: type  # Python type (float, int, str, etc.)
    source: str  # Provider name
    description: str  # Human-readable description
    category: Optional[str] = (
        None  # Feature category (price, technical, economic, etc.)
    )

    def __post_init__(self):
        """Validate feature data"""
        if not self.name:
            raise ValueError("Feature name cannot be empty")
        if not self.source:
            raise ValueError("Feature source cannot be empty")
